client/src/components/molecules/EditForm/SizeSelector.js

```diff
@@ -20,33 +21,47 @@ const SizeSelector = ({
 
   useEffect(() => {
     // clear quantity values when category is changed
-    setFieldValue(`shoeSizeBatchValues`, {});
-    setFieldValue('clothingSizeBatchValues', {});
+    setFieldValue(`shoeSizes`, {});
+    setFieldValue('clothingSizes', {});
   }, [category, subcategory, setFieldValue]);
 
   const handleQuantityChange = (size, quantity) => {
     const validQuantity = Math.max(quantity, 0);
     const updatedQuantities = {
-      ...formikProps.values[`${fieldName}BatchValues`],
+      ...formikProps.values[`${fieldName}`],
       [size]: validQuantity,
     };
-    setFieldValue(`${fieldName}BatchValues`, updatedQuantities);
+
+    // perform a sort on the selected sizes as this will be important when they are displayed.
+    let sizeOrder = [];
+    if (fieldName === 'shoeSizes') {
+      sizeOrder = shoeSizeOptions;
+    } else {
+      sizeOrder = clothingSizeOptions;
+    }
+    const keyValueArray = Object.entries(updatedQuantities);
+    keyValueArray.sort(
+      (a, b) => sizeOrder.indexOf(a[0]) - sizeOrder.indexOf(b[0])
+    );
+    const sortedUpdatedQuantities = Object.fromEntries(keyValueArray);
```

Comment:
Good point - I think it's pretty negligible given the size of the array is quite small, but I think your comment made me realize I can just as easily have this done once on the backend as the data is sent through (1 single sort) instead of having it being sorted each time a number is added - small fix, that I'll do once I push this through (to get the testing started).